TypeCobol.Test/Utils/TestUnit.cs

```diff
@@ -0,0 +1,152 @@
+            foreach (var copyDirectory in copyDirectories)
+            {
+                compilationProject.SourceFileProvider.AddLocalDirectoryLibrary(copyDirectory, false, copyExtensions, documentFormat);
+            }
+
+            bool isCopyFile = copyExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase);
+            _fileCompiler = new FileCompiler(compilationProject, fileNameWithoutExtension, isCopyFile);
+
+            if (antlrProfiling)
+            {
+                _fileCompiler.CompilationResultsForProgram.PerfStatsForCodeElementsParser.ActivateDetailedAntlrPofiling = true;
+                _fileCompiler.CompilationResultsForProgram.PerfStatsForTemporarySemantic.ActivateDetailedAntlrPofiling = true;
+            }
+
+            _inputChanges = new List<InputChange>();
+            _initialResultComparisons = new List<Comparison>();
+            _intermediateResultsComparisons = new Dictionary<string, List<Comparison>>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        public void AddInputChange(InputChange inputChange)
+        {
+            _inputChanges.Add(inputChange);
+        }
+
+        public void AddComparison(Comparison comparison)
+        {
+            if (comparison.ChangeId == null)
+            {
+                // Initial comparison
+                _initialResultComparisons.Add(comparison);
+            }
+            else
+            {
+                // Intermediate comparison
+                if (!_intermediateResultsComparisons.TryGetValue(comparison.ChangeId, out var comparisons))
+                {
+                    comparisons = new List<Comparison>();
+                    _intermediateResultsComparisons.Add(comparison.ChangeId, comparisons);
+                }
+
+                comparisons.Add(comparison);
+            }
+        }
+
+        public void Run()
+        {
+            // Pre-conditions
+#if EUROINFO_RULES
+            RemoveNonEIComparisons(_initialResultComparisons);
+            foreach (var intermediateComparisons in _intermediateResultsComparisons.Values)
+            {
+                RemoveNonEIComparisons(intermediateComparisons);
+            }
+
+            void RemoveNonEIComparisons(List<Comparison> comparisons)
+            {
+                if (comparisons.Any(c => c.IsEI))
+                {
+                    // If any -EI result file exists => Remove all comparators without isEI flag to true. 
+                    // We only want to check EI results files. 
+                    comparisons.RemoveAll(c => !c.IsEI);
+                }
+            }
+#endif
+#if DEBUG
+            // Incremental changes must appear in .inc file in same order as result files on disk
+            var changes = _inputChanges.Select(change => change.Id); // Order of changes to be applied
+            var results = _intermediateResultsComparisons.Keys; // Order of intermediate result files on disk
```

Comment:
For me, the order of keys in a Dictionnary is undefined. A List should be used to check the order.
And the order of files returned by `Directory.EnumerateFiles` is also not guaranteed (https://stackoverflow.com/questions/14097264/directory-enumeratefiles-read-order).

So maybe instead, we should only check that changes ID are sorted:
```C#
            if (_inputChanges.SequenceEqual(_inputChanges.OrderBy(c => c.Id)))
            {
                throw new Exception("Incremental change must be sorted");
            }
```

Reply:
ok done